src/legacy/ui/public/visualize/loader/pipeline_helpers/build_pipeline.ts

```diff
@@ -247,41 +281,55 @@ export const buildPipelineVisFunction: BuildPipelineVisFunction = {
     return `kibana_table ${prepareJson('visConfig', visConfig)}`;
   },
   metric: (visState, schemas) => {
-    const visConfig = {
-      ...visState.params,
-      ...buildVisConfig.metric(schemas),
-    };
-    return `kibana_metric ${prepareJson('visConfig', visConfig)}`;
+    const {
+      percentageMode,
+      useRanges,
+      colorSchema,
+      metricColorMode,
+      colorsRange,
+      labels,
+      invertColors,
+      style,
+    } = visState.params.metric;
+    const { metrics, bucket } = buildVisConfig.metric(schemas).dimensions;
+
+    let expr = `metricvis `;
+    expr += prepareValue('percentage', percentageMode);
+    expr += prepareValue('colorScheme', colorSchema);
+    expr += prepareValue('colorMode', metricColorMode);
+    expr += prepareValue('useRanges', useRanges);
+    expr += prepareValue('invertColors', invertColors);
+    expr += prepareValue('showLabels', labels && labels.show);
+    if (style) {
+      expr += prepareValue('bgFill', style.bgFill);
+      expr += prepareValue('font', `{font size=${style.fontSize}}`, true);
+      expr += prepareValue('subText', style.subText);
+      expr += prepareDimension('bucket', bucket);
+    }
+
+    if (colorsRange) {
+      colorsRange.forEach((range: any) => {
```

Comment:
instead of `range: any` you should be able to import the Range type from the interpreter to use here

Reply:
i didn't want to do this, not to import types from other plugins (which can be disabled)

Reply:
the interfaces should still be importable even if the plugin is disabled since they are static though, right?

Reply:
we should not rely on that, plugin could also be deleted.